server/src/main/java/io/unitycatalog/server/service/TableService.java

```diff
@@ -2,55 +2,168 @@
 import io.unitycatalog.server.model.CreateTable;
+import io.unitycatalog.server.model.ListTablesResponse;
+import io.unitycatalog.server.model.SchemaInfo;
 import io.unitycatalog.server.model.TableInfo;
+import io.unitycatalog.server.persist.CatalogRepository;
+import io.unitycatalog.server.persist.MetastoreRepository;
+import io.unitycatalog.server.persist.SchemaRepository;
 import io.unitycatalog.server.persist.TableRepository;
+import io.unitycatalog.server.persist.model.Privileges;
+import io.unitycatalog.server.utils.IdentityUtils;
+import lombok.SneakyThrows;
+
+import java.util.List;
+import java.util.Map;
 import java.util.Optional;
+import java.util.UUID;
+
+import static io.unitycatalog.server.model.SecurableType.CATALOG;
+import static io.unitycatalog.server.model.SecurableType.METASTORE;
+import static io.unitycatalog.server.model.SecurableType.SCHEMA;
+import static io.unitycatalog.server.model.SecurableType.TABLE;
 
 @ExceptionHandler(GlobalExceptionHandler.class)
 public class TableService {
 
   private static final TableRepository TABLE_REPOSITORY = TableRepository.getInstance();
+  private static final SchemaRepository SCHEMA_REPOSITORY = SchemaRepository.getInstance();
+  private static final CatalogRepository CATALOG_REPOSITORY = CatalogRepository.getInstance();
 
-  public TableService() {}
+  private final UnityCatalogAuthorizer authorizer;
+  private final UnityAccessEvaluator evaluator;
+
+  @SneakyThrows
+  public TableService(UnityCatalogAuthorizer authorizer) {
+    this.authorizer = authorizer;
+    evaluator = new UnityAccessEvaluator(authorizer);
+  }
 
   @Post("")
-  public HttpResponse createTable(CreateTable createTable) {
+  // The Databricks API does not have a create table endpoint, so defining sensible rules here.
+  @AuthorizeExpression("""
+      #authorize(#principal, #metastore, OWNER) ||
+      #authorizeAny(#principal, #catalog, OWNER, USE_CATALOG) && #authorizeAny(#principal, #schema, OWNER, USE_SCHEMA)
+      """)
+  @AuthorizeKey(METASTORE)
+  public HttpResponse createTable(
+      @AuthorizeKeys({
+            @AuthorizeKey(value = SCHEMA, key = "schema_name"),
+            @AuthorizeKey(value = CATALOG, key = "catalog_name")
+          })
+          CreateTable createTable) {
     assert createTable != null;
-    TableInfo createTableResponse = TABLE_REPOSITORY.createTable(createTable);
-    return HttpResponse.ofJson(createTableResponse);
+    TableInfo tableInfo = TABLE_REPOSITORY.createTable(createTable);
+    initializeAuthorizations(tableInfo);
+    return HttpResponse.ofJson(tableInfo);
   }
 
   @Get("/{full_name}")
-  public HttpResponse getTable(@Param("full_name") String fullName) {
+  @AuthorizeExpression("""
+          #authorize(#principal, #metastore, OWNER) ||
+          (#authorizeAny(#principal, #schema, OWNER, USE_SCHEMA) && #authorizeAny(#principal, #catalog, OWNER, USE_CATALOG) && #authorizeAny(#principal, #table, OWNER, SELECT))
```

Comment:
other privileges that allow getTable:
- catalog owner
- schema owner with USE_CATALOG
- principal with USE_CATALOG, USE_SCHEMA, and any privilege on the table